sklearn/ensemble/_hist_gradient_boosting/tests/test_gradient_boosting.py

```diff
@@ -744,3 +749,50 @@ def test_single_node_trees(Est):
 def test_custom_loss(Est, loss, X, y):
     est = Est(loss=loss, max_iter=20)
     est.fit(X, y)
+
+
+@pytest.mark.parametrize('HistGradientBoosting, X, y', [
+    (HistGradientBoostingClassifier, X_classification, y_classification),
+    (HistGradientBoostingRegressor, X_regression, y_regression),
+    (HistGradientBoostingClassifier,
+        X_multi_classification, y_multi_classification),
+])
+def test_staged_predict(HistGradientBoosting, X, y):
+
+    # Test whether staged predictor eventually gives
+    # the same prediction.
+    X_train, X_test, y_train, y_test = train_test_split(
+        X, y,
+        test_size=0.5,
+        random_state=0
+    )
+    gb = HistGradientBoosting(max_iter=10, random_state=0)
+
+    # test raise NotFittedError if not fitted
+    with pytest.raises(NotFittedError):
+        next(gb.staged_predict(X_test))
+
+    gb.fit(X_train, y_train)
+
+    # test if the staged predictions of each iteration
+    # are equal to the corresponding predictions of the same estimator
+    # trained from scratch.
+    # this also test limit case when max_iter = 1
+    method_names = (
+        ['predict'] if is_regressor(gb)
+        else ['predict', 'predict_proba', 'decision_function']
+    )
+    for method_name in method_names:
+        method = getattr(gb, method_name)
```

Comment:
you shoudn't need to use `getattr(gb, method_name)`

we want to compare the output of gb.staged_method with the output of aux.method

Reply:
I used ```getattr(gb, method_name)``` to compare the final output of staged method is equal to the output of regular method. Or should it be removed?